tools/sdk-generation-pipeline/packages/sdk-generation-lib/src/lib/runScript.ts

```diff
@@ -1,10 +1,10 @@
-import {RunOptions} from "../types/taskInputAndOuputSchemaTypes/CodegenToSdkConfig";
-import * as path from "path";
-import {spawn} from "child_process";
-import {logger} from "../utils/logger";
-import {Readable} from "stream";
-import {scriptRunningState} from "../types/scriptRunningState";
-import * as fs from "fs";
+import { RunOptions } from '../types/taskInputAndOuputSchemaTypes/CodegenToSdkConfig';
+import * as path from 'path';
+import { spawn } from 'child_process';
+import { logger } from '../utils/logger';
+import { Readable } from 'stream';
+import { scriptRunningState } from '../types/scriptRunningState';
+import * as fs from 'fs';
 
 export const isLineMatch = (line: string, filter: RegExp | undefined) => {
```

Comment:
Can you help me understand what this runScript code is used for? What sort of scripts to we intend to be running? Why are we not just using a shell script to execute these?

Reply:
hi @dw511214992  dose sdk generation pipeline docker still need runScript? If so, can you help me answer the questions, thx.

Reply:
runScript is used for run the customized script, such as init script, generateAndBuld script, mockTest script. It supports all kinds of script os supported, which means it can be `shell` script, `powershell` script, or binary files. We supports so many kinds of script because sdk owners wirte automation script in different ways.
Also, we cannot exec script with shell directly because we need to listen the logs outputed by `script` and the listened logs are processed or write to files/console.

Reply:
When you say listen to log outputs what do you mean? You can definitely redirect the outputs in a shell script to capture the output as needed. I realize that you are mimicking the existing runScript.ts but I'd like for use to slowly move to using more standard things like calling scripts from shell instead of having to handle all that inside of this TS script.

Reply:
> When you say listen to log outputs what do you mean? You can definitely redirect the outputs in a shell script to capture the output as needed. I realize that you are mimicking the existing runScript.ts but I'd like for use to slowly move to using more standard things like calling scripts from shell instead of having to handle all that inside of this TS script.

I agree with you, and I will discuss with Wei if we can improve it in the future version